### PR TITLE
[fixes #2991] works in iojs now.

### LINKED ISSUE #2991: io.js support
We should test compatibility with io.js and set up ci to test it as well :)

### PR DESCRIPTION
**Pull request by stefanpenner**
Turns out ncp, currently doesn’t work in iojs, so we moved to node-copy-dereference

relevant ncp issue: https://github.com/AvianFlu/ncp/issues/79

- [x] make tests use node-copy-dereference
- [x] ~~add iojs to travis (if it is reasonably supported)~~ pendinghttps://github.com/travis-ci/travis-ci/issues/3108
- [x] tests green

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -155,6 +154,7 @@
     "mocha": "^1.18.0",
     "mocha-jshint": "0.0.9",
     "multiline": "^1.0.1",
+    "ncp": "^1.0.1",
```
Comment (rwjblue) [inline package.json]: We should remove this, I think.

Comment (rwjblue) [inline package.json]: It is only used in one other location (test fixture copying).

Comment (stefanpenner) [inline package.json]: ya its already a task item above

Comment (rwjblue) [inline package.json]: LOL, I guess I should read the descriptions every once in a while ;)

File: tests/helpers/copy-fixture-files.js [hunk 2]
```diff
@@ -2,13 +2,12 @@
 
 var path    = require('path');
 var Promise = require('../../lib/ext/promise');
-var ncp     = Promise.denodeify(require('ncp'));
+var cpd     = require('ember-cli-copy-deference');
```
Comment (rwjblue) [inline tests/helpers/copy-fixture-files.js]: Typo.  Should be `ember-cli-copy-dereference`.

File: lib/models/builder.js [hunk 3]
```diff
@@ -4,10 +4,11 @@ var fs          = require('fs');
 var path        = require('path');
 var Promise     = require('../ext/promise');
 var rimraf      = Promise.denodeify(require('rimraf'));
-var ncp         = Promise.denodeify(require('ncp'));
 var Task        = require('./task');
 var SilentError = require('../errors/silent');
 var chalk       = require('chalk');
+var cpd         = require('ember-cli-copy-dereference');
+var mkdirp      = require('mkdirp');
```
Comment (chadhietala) [inline lib/models/builder.js]: `fs-extra` is a dependency which has `fs.mkdirs`. It does `mkdir -p` by default.

Comment (stefanpenner) [inline lib/models/builder.js]: im out of time currenty, do you mind submitting a PR to my PR with the fix?

Comment (chadhietala) [inline lib/models/builder.js]: Yea I'll PR fixing it across the project.

File: tests/helpers/copy-fixture-files.js [hunk 4]
```diff
@@ -2,13 +2,12 @@
...
+var cpd     = require('ember-cli-copy-dereference');
 
 var rootPath = process.cwd();
 
 module.exports = function copyFixtureFiles(sourceDir) {
-  return ncp(path.join(rootPath, 'tests', 'fixtures', sourceDir), '.', {
-    clobber: true,
-    stopOnErr: true
+  return new Promise(function(resolve) {
+    resolve(cpd(path.join(rootPath, 'tests', 'fixtures', sourceDir), '.'));
```
Comment (rwjblue) [inline tests/helpers/copy-fixture-files.js]: should be `cpd.sync` instead of `cpd`

**Comment by stefanpenner** [comment]
i think the latest failures are NCP clobbers, so we need to rimraf before, or...

**Comment by stefanpenner** [comment]
appveyor is incorrect, https://ci.appveyor.com/project/embercli/ember-cli/build/746 is still running.

**Comment by rwjblue** [comment]
@stefanpenner - Says it was canceled if you review the status at that URL.

**Comment by rwjblue** [comment]
screenshot

**Comment by rwjblue** [comment]
restarting

**Comment by rwjblue** [comment]
Apparently, I am crazy.  Stef is of course correct, but the Travis commit status is incorrect.

**Comment by stefanpenner** [comment]
ITS GREEN!

**Comment by jayphelps** [comment]
ITS SO BEAUTIFUL

**Pull request merged**